src/freenet/node/RequestHandler.java

```diff
@@ -777,7 +777,7 @@ private void finishOpennetInner(OpennetManager om) {
 			noderef = rs.waitForOpennetNoderef();
 		} catch (WaitedTooLongForOpennetNoderefException e) {
 			sendTerminal(DMT.createFNPOpennetCompletedTimeout(uid));
-			rs.ackOpennet(rs.successFrom());
+			// RequestSender will have already acknowledged.
```

Comment:
This would be a good place to add an assertion on a package-private getter for `RequestSender.opennetFinishedRelaying`.